sdk/core/Azure.Core/tests/ModelSerializationTests/Animal.cs

```diff
@@ -0,0 +1,159 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+
+namespace Azure.Core.Tests.ModelSerializationTests
+{
+    public class Animal : IJsonSerializable, IUtf8JsonSerializable
```

Comment:
Is the intended goal that Azure SDK public models will implement both of these interfaces?  Will `IUtf8JsonSerializable` remain internal?  Can you help us understand the thinking around this?

Reply:
I believe that the intended goal is to expose TrySerialize and TryDeserialize in IJsonSerializable to all users. I think the goal is to generate these two methods for all public models as currently users cannot serialize/ deserialize on their own as these methods are internal. IUtf8JsonSerializable would remain internal and needs to be updated - we are just using a copy for now.

Reply:
We didn't want to gut the underlying serialization implementation just yet.  Our first step here is create this public interface into what is already happening and this gives us the flexibility to adjust the internal implementation whenever we want as long as functionally it performs the same.

Reply:
I see.  For building out our conceptual models, does this mean we should think of IJsonSerializable as an eventual replacement for IUtf8JsonSerializable where it appears in Core APIs (although the practical implementation will be staged in steps for convenience)?

Reply:
This is a great question. Having 2 interfaces gives us the flexibility but it also increases the burden of keeping them consistent.